extensions/vscode/src/multiStepInputs/newCredential.ts

```diff
@@ -75,10 +76,17 @@ export async function newCredential(
   // ***************************************************************
   async function inputServerUrl(input: MultiStepInput, state: MultiStepState) {
     const thisStepNumber = assignStep(state, "inputServerUrl");
-    const currentURL =
+    let currentURL =
       typeof state.data.url === "string" && state.data.url.length
         ? state.data.url
         : "";
+
+    if (currentURL === "") {
+       const configURL: string | null = getXDGConfigProperty("rstudio/rsession.conf", "default-rsconnect-server");
```

Comment:
It looks like this section only got the `XDGConfigProperty` treatment did you mean to include the `currentURL = extensionSettings.defaultConnectServer();` checking here as well? It does seem like a good spot for it.

Even better could provide a single function that either returns the `defaultConnectServer` or the XDG Config Property depending for both of these to use.

Reply:
Great catch and suggestion, thanks!